src/main/java/org/eclipse/golo/runtime/ClassReferenceSupport.java

```diff
@@ -55,7 +55,17 @@ public static CallSite bootstrap(MethodHandles.Lookup caller, String name, Metho
     try {
       return Class.forName(name, true, classLoader);
     } catch (ClassNotFoundException e) {
-      return null;
+      switch(name) {
```

Comment:
Another option is to avoid a map allocation (even if it's small) and refactor that `switch` to a `private` method. That's perhaps even more elegant.